### PR TITLE
Pal 389 user service springboot cacheable

### PR DESCRIPTION
**Pull request by developer01189998819991197253**
Deleted old cache mechanism, removed backingStores
Added Caffeine cache
Updated Copyright
Added CachedUserService for @Cacheable method
Removed CompletableFuture from userService 
Removed cache tests and have added them to Integration tests
Added UserControllerTest

### CODE REVIEW

**Review by pd104923 — DISMISSED** [review]

**Review by dev-958 — CHANGES_REQUESTED** [review]
Needs some redesign to the structure, spoke to Chris.

**Review by dev-958 — DISMISSED** [review]

File: user-service/src/main/resources/application.yaml [hunk 1]
```diff
@@ -13,6 +13,9 @@
 # limitations under the License.
 #
 spring:
+  cache:
+    caffeine:
+      spec: expireAfterWrite=60s, maximumSize=500
```
Comment (ac74475) [inline user-service/src/main/resources/application.yaml]: If this is the default config then doesn't this mean that if you have more than 60 seconds between configuring the example and running it, then you will hit user not found exceptions as we are using the null-user-service

Comment (ac74475) [reply user-service/src/main/resources/application.yaml]: If my above assumption is correct then I suggest making the ttl 1hour

**Review by ac74475 — CHANGES_REQUESTED** [review]
Why do we have an empty users.txt file?
Also there is no need to update the copyright year on each file unless you have modified the majority of the file.

File: user-service/src/main/resources/application.yaml (hunk 1 above)
Comment (developer01189998819991197253) [reply user-service/src/main/resources/application.yaml]: agreed to set ttl to 60m

**Review by ac74475 — APPROVED** [review]

**Pull request merged**